lib/todolist.ex

```diff
@@ -35,8 +35,8 @@ defmodule TodoList do
       iex> TodoList.new() |> TodoList.add_entry(%{date: ~D[2021-12-19], title: "job"})
       %TodoList{auto_id: 2, entries: %{1 => %{date: ~D[2021-12-19], id: 1, title: "job"}}}
   """
-  @spec add_entry(TodoList.t(), map) :: TodoList.t()
-  def add_entry(todo_list, entry) do
+  @spec add_entry(t(), entry()) :: t()
+  def add_entry(%TodoList{} = todo_list, entry) do
```

Comment:
The good practice is to accept only what is expected; that way an attempt to pass a list/atom/whatever there will fail with a well understandable error.